yacc/reader.c

```diff
@@ -316,7 +315,6 @@ static void process_open_curly_bracket(FILE *f) {
                     size_t i;
                     for (i = 0; i <= size; ++i) {
                         if (cptr[i] != buf[i]) {
-                            newcptr--;
```

Comment:
It is not obvious how this change relates to the PR. (Am I missing something subtle that makes this change necessary?). If I understand correctly, it was unnecessary from the start and you are just combining a small cleanup with the patch. Can you confirm?